source/modulo_utils/modulo_utils/binary_io.py

```diff
@@ -0,0 +1,119 @@
+import os
+import struct
+import time
+from datetime import datetime
+from io import BufferedWriter
+from typing import List
+
+import clproto
+from modulo_interfaces.msg import EncodedState
+
+
+class BinaryRecorder():
```

Comment:
hm, how would be feel about calling this something more specific?

```suggestion
class EncodedStateRecorder():
```